src/main/java/org/logstash/beats/Batch.java

```diff
@@ -0,0 +1,47 @@
+package org.logstash.beats;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class Batch {
+    private byte protocol = Protocol.VERSION_2;
+    private int windowSize;
```

Comment:
I think we should keep consistent naming around `Batch` and `Window`. We seem to say that Batches have a window size. Since windows are commonly used to describe overlapping batches, I think I'd prefer the term Batch over Window. so maybe we can rename `windowSize` to `batchSize`?